proto/test/v1/conformance-test.proto

```diff
@@ -0,0 +1,71 @@
+syntax = "proto3";
+
+package google.api.expr.test.v1;
+
+import "google/api/expr/v1alpha1/checked.proto";
+import "google/api/expr/v1alpha1/eval.proto";
+import "google/api/expr/v1alpha1/value.proto";
+import "google/protobuf/empty.proto";
+
+// Configuration needed for running any test.
+message Config {
+  // Command line for executing the CelService server to be used
+  // when no call-specific server is specified.
+  string server_cmd = 1;
+
+  // Command line for executing a CelService server to be used
+  // for Parse() calls.  Default to server_cmd if empty.
+  string parse_server_cmd = 2;
+
+  // Command line for executing a CelService server to be used
+  // for Check() calls.  Default to server_cmd if empty.
+  string check_server_cmd = 3;
+
+  // Command line for executing a CelService server to be used
+  // for Eval() calls.  Default to server_cmd if empty.
+  string eval_server_cmd = 4;
+}
+
+message EvalResponseMatcher {
+  oneof kind {
+    google.api.expr.v1alpha1.Value value = 1;
+    ErrorSets errors = 2;  // expect one of these
+    UnknownSets unknowns = 3;  // expect one of these
+    string parse_failure_regex = 4;  // "." until we standardize
+    string check_failure_regex = 5;
+    google.protobuf.Empty trueval = 6; // shortcut for true boolean value
+  }
+}
+
+message ErrorSets {
+  // Success if we match any of these sets.
+  repeated google.api.expr.v1alpha1.ErrorSet errors = 1;
+}
+
+message UnknownSets {
+  // Success if we match any of these sets.
+  repeated google.api.expr.v1alpha1.UnknownSet unknowns = 1;
+}
+
+message SimpleEvalTest {
+  string name = 1;
+  string description = 2;
+  string expr = 3;
+  repeated google.api.expr.v1alpha1.Decl type_env = 4;
```

Comment:
As a future TODO, consider creating separate messages for parse / check / eval to encapsulate the different parameters. e.g.

```
message ConformanceTestCase {
  message TestExpr {
    oneof kind {
      string expr = 1;
      google.api.expr.v1alpha1.Expr expr_ast = 2;
    }
  }

  message Parse {
    bool disable_macros = 1;
    ParseMatcher matchers = 2;
  }

  message Check {
    repeated google.api.expr.v1alpha1.Decl declarations = 1;
    repeated FileDescriptorSet types = 2;
    repeated CheckMatcher matchers = 3;
  }

  message Eval {
     map<string, google.api.expr.v1alpha1.ExprValue> bindings = 1;
     repeated EvalMatcher matchers = 2;
  }

  string name = 1;
  string description = 2;
  TestExpr test_expr = 3;
  Parse parse = 4;
  Check check = 5;
  Eval eval = 6;
}
```

Reply:
I think I better understand what you're aiming for here, now that I've grok'd the config section and how it could be used to mix and match different implementations of the CEL stack together.

Maybe this?

```proto
message EvalTest {
  message ResultMatcher {
    oneof kind {
       UnknownSetMatcher any_unknowns = 2;
    }
  }

  string name = 1;
  string description = 2;
  string expr = 3;
  bool disable_macros = 4;
  repeated google.api.expr.v1alpha1.Decl env = 5;
  map<string, google.api.expr.v1alpha1.ExprValue> bindings = 6;
  
  // static_errors, checked_type, and result field output should be ANDed together.
  repeated string static_errors = 7
  google.api.expr.v1alpha1.Type checked_type = 8
  oneof result {
    ResultMatcher matcher = 9
    // You might need non-exact matching for maps, could be in the ResultMatcher.
    google.api.expr.v1alpha1.Value value = 10;
    ErrorSet runtime_errors = 11;
    UnknownSet unknowns = 12;
    ErrorSetMatcher any_runtime_errors = 13;
    UnknownSetMatcher any_unknowns = 14;
   }
}
```

Here are a couple of example cases
```textproto
eval_test {
  name: "one plus one"
  expr: "1 + 1"
  checked_type { primitive: Type.INT64 }
  value { int64_value: 2 }
}

eval_test {
  name: "one div zero"
  expr: "1 / 0"
  checked_type { primitive: Type.INT64 }
  runtime_error { message: "divide by zero." }
}

eval_test {
  name: "one div zero and uint overflow"
  expr: "1 / 0 && uint(-1) != 1u"
  // implied checked_type of bool
  any_runtime_errors {
     runtime_error { message: "divide by zero." }
     runtime_error { message: "overflow." }
   }
}
```

Reply:
I'm fine with an implicit `type` check against the `value`. I think it should probably also be present whenever the `value` is not specified, but otherwise that sounds good.